kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -145,8 +143,14 @@ public Snapshot getSnapshotAt(
      */
     public Snapshot getSnapshotForTimestamp(
             TableClient tableClient, long millisSinceEpochUTC) throws TableNotFoundException {
+        long startTimeMillis = System.currentTimeMillis();
```

Comment:
considered using an utility method similar to Scala (e.g. withWalltimeMeasured), but the issue with in Java is `Supplier`s don't allow throwing exceptions and the code is becoming even more complicated.